scripts/solve.js

```diff
@@ -1,7 +1,33 @@
+const CODE_AREA_ID      = 'code-area';
+const CODE_OUTPUT_ID    = 'code-output';
+const NEWLINE           = '\n';
+
+var codeMirror;
+var outputArea;
+
+
 window.addEventListener('load', function() {
-  var myCodeMirror = CodeMirror(document.getElementById('code-area'), {
-    value: "function myScript(){return 100;}\n",
-    mode:  "javascript",
-    lineNumbers: true
+  codeMirror = CodeMirror(document.getElementById(CODE_AREA_ID), {
+    value: 'function myScript(){return 100;}\n',
+    mode:  'javascript',
+    lineNumbers: true,
   });
+
+  outputArea = document.getElementById(CODE_OUTPUT_ID);
 });
+
+function overrideFunctions(interpreter, scope) {
```

Comment:
is this something that you only want in the prototype? Or are users supposed to use alert() and assert() to debug in the application? If so, that is a bit unintuitive to me and should be documented when we move on from the prototype.

Reply:
Thanks for the feedback!

>is this something that you only want in the prototype? Or are users supposed to use alert() and assert() to debug in the application?

It is something that I think should be kept in the application. We plan to add step-by-step debugging if we have enough time, but I think being able to do basic printing and assertion on top of that would be useful for users in general. 

We could also use this to extend the application after the MVP. I'm currently thinking of something along the line of making tests with x% coverage, for example.

>If so, that is a bit unintuitive to me and should be documented when we move on from the prototype.

I intend to document this in the user interface as me move into the MVP. I'm currently thinking of something like a `?` button in the UI that will explain the usage of these functions when the user hover their mouse over it.

Regarding it being unintuitive, do you have any suggestions on what would be a more intuitive design?

Reply:
I think it should be fine as long as they are documented in the UI as you said.